### PR TITLE
Added sarsen and xarray-sentinel

### PR DESCRIPTION
**Pull request by TomAugspurger**
This adds sarsen, and its dependency xarray-sentinel.

Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [ ] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [ ] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by TomAugspurger** [comment]
@aurghs and @alexamici can you confirm you're OK with being added as maintainers for these?

**Comment by TomAugspurger** [comment]
I'm not immediately sure why CI is failing. According to https://conda-forge.org/docs/maintainer/adding_pkgs.html#adding-multiple-packages-at-once you can add multiple recipes at once, but I haven't done that before.

I see references to both `xarray-sentinel-0.0.0` and `xarray-sentinel-0.4.2` in the build logs. I'd think that the one that matters is at

```
  + xarray-sentinel                   0.4.2  pyhd8ed1ab_0         /home/conda/staged-recipes/build_artifacts/noarch       54kB

```

https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=490989&view=logs&j=6f142865-96c3-535c-b7ea-873d86b887bd&t=22b0682d-ab9e-55d7-9c79-49f3c3ba4823&l=1394

in the plan for building `sarsen`.

But the pip check fails:

```

+ pip check
sarsen 0.0.0 has requirement xarray-sentinel>=0.4, but you have xarray-sentinel 0.0.0.
```

**Comment by alexamici** [comment]
@TomAugspurger great work!

I confirm I'm willing to maintain the recipe.

**Comment by aurghs** [comment]
Thank you very much! 
I am also willing to maintain the recipes.

File: recipes/xarray-sentinel/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+requirements:
+  host:
+    - pip
+    - python >=3.8
+    - setuptools_scm
+  run:
+    - attrs
+    - dask-core
+    - fsspec
+    - python >=3.7
```
Comment (ocefpaf) [inline recipes/xarray-sentinel/meta.yaml:27]: Missed this one but it should be the same version as the one in host.

Comment (TomAugspurger) [reply recipes/xarray-sentinel/meta.yaml:27]: Whoops. I'll fix it in the feedstock.

**Pull request merged**